docs/source/developers/java/building.rst

```diff
@@ -347,6 +347,11 @@ Arrow repository, and update the following settings:
 * If using IntelliJ's Maven integration to build, you may need to change
   ``<fork>`` to ``false`` in the pom.xml files due to an `IntelliJ bug
   <https://youtrack.jetbrains.com/issue/IDEA-278903>`__.
+* To enable debugging JNI-based modules like ``dataset``,
+  activate specific profiles in the Maven tab under "Profiles".
+  Ensure the profiles ``arrow-c-data``, ``arrow-jni``, ``generate-libs-cdata-all-os``,
+  ``generate-libs-jni-macos-linux``, and ``jdk11+`` are enabled, which equips the IDE
+  with the necessary debugging utilities for test classes.
```

Comment:
nit: it's just telling IntelliJ to build them in the first place, otherwise they're ignored by our Maven config

Reply:
@lidavidm I updated the wording. Does it look okay?